src/graph.py

```diff
@@ -6,23 +6,27 @@
 class Graph(object):
 
     def __init__(self, initial_graph=None):
-        '''Pass in a graph in the form of a dictionary with the nodes as
-        key, and a list of edges as value, for example
-        {1:[2, 3]},
-         2:[]},
-         3:[1]}
+        '''Pass in a graph in the form of a dictionary with the nodes as 
+        key, and a dictoinary containing edges as keys and wieghts as values
+        {1:{2:100, 3:299},
+         2:{},
+         3:{1:150}}
         '''
         if initial_graph is None:
-            self.graph = {}
-        else:
+                self.graph = {}
+        elif isinstance(initial_graph, dict):
             self.graph = initial_graph
+        else:
+            raise TypeError("Please pass a dictionary")
 
     def __iter__(self):
         return iter(self.graph)
 
     def nodes(self):
         '''return a list of all nodes in the graph'''
-        return self.graph.keys()
+        # return self.depth_first_traversal(self.graph)
+
+        return list(self.graph.keys())
 
     def edges(self):
         '''returns a list of tuples, that are edges'''
```

Comment:
Currently your `edges` method returns a generator of dictionaries. You need to either update your docstring for the `edges` method, or change your `edges` method so it once again returns a list of tuples. I recommend the latter.